sphinx/ext/graphviz.py

```diff
@@ -252,6 +260,9 @@ def render_dot(self, code, options, format, prefix='graphviz'):
         # Graphviz may close standard input when an error occurs,
         # resulting in a broken pipe on communicate()
         stdout, stderr = p.communicate(code)
+        # print stderr if it is there
+        if stderr:
+            logger.warning(stderr.decode())
```

Comment:
We already show stderr message if exist status is non-zero.
https://github.com/sphinx-doc/sphinx/blob/f697717c0e4870a2f5dd99dba42c785f10595db4/sphinx/ext/graphviz.py#L262-L264

My concerns are following:

* I don't know the message on that situation is important for users.
* The message is shown twice if dot command exists with non-zero.
* The message is shown without any headers. So users might not understand where the message comes from.

>Perhaps it could be info instead of warning?

As you said, this is "errors/wranings from dot". So WARNING level is better to me if we emits this.

Reply:
Right, I see your point. I suppose for debugging the user could always copy whatever is inside the graphviz directive to a file and manually run dot. My intention was to remove the need for this extra step, but it's a bit of a hassle to make it work nicely within this context. I'll remove it.